### ISSUE TITLE
"istio-ca-root-cert" failure in multicluster tests

### ISSUE THREAD

**Issue by howardjohn** [istio/istio#29419]
https://storage.googleapis.com/istio-prow/logs/integ-pilot-multicluster-tests_istio_postsubmit/1334950282827665408/build-log.txt

**Comment by howardjohn** [comment]
`2020-12-04T20:06:17.671573Z	error	klog	error retrieving resource lock istio-system/istio-namespace-controller-election: configmaps "istio-namespace-controller-election" is forbidden: User "system:serviceaccount:istio-system:istio-reader-service-account" cannot get resource "configmaps" in API group "" in the namespace "istio-system"[]` in logs... slow API server?

also a bunch of `http: TLS handshake error from 10.10.0.10:55008: remote error: tls: bad certificate` which I think is the api server, probably doesn't have caBundle patched?


filtered logs:
```
2020-12-04T20:06:08.097884Z     info    joining leader-election for istio-namespace-controller-election in istio-system
2020-12-04T20:06:08.098317Z     info    klog    attempting to acquire leader lease  istio-system/istio-namespace-controller-election...[]
2020-12-04T20:06:17.652468Z     info    joining leader-election for istio-namespace-controller-election in istio-system
2020-12-04T20:06:17.652570Z     info    klog    attempting to acquire leader lease  istio-system/istio-namespace-controller-election...[]
2020-12-04T20:06:17.671573Z     error   klog    error retrieving resource lock istio-system/istio-namespace-controller-election: configmaps "istio-namespace-controller-election" is forbidden: User "system:serviceaccount:istio-system:istio-reader-service-account" cannot get resource "configmaps" in API group "" in the namespace "istio-system"[]
2020-12-04T20:06:21.756629Z     info    joining leader-election for istio-namespace-controller-election in istio-system
2020-12-04T20:06:21.757067Z     info    klog    attempting to acquire leader lease  istio-system/istio-namespace-controller-election...[]
2020-12-04T20:06:21.859424Z     info    joining leader-election for istio-namespace-controller-election in istio-system
2020-12-04T20:06:21.859793Z     info    klog    attempting to acquire leader lease  istio-system/istio-namespace-controller-election...[]
2020-12-04T20:06:21.962684Z     info    joining leader-election for istio-namespace-controller-election in istio-system
2020-12-04T20:06:21.962816Z     info    klog    attempting to acquire leader lease  istio-system/istio-namespace-controller-election...[]
2020-12-04T20:06:26.602434Z     info    klog    successfully acquired lease istio-system/istio-namespace-controller-election[]
2020-12-04T20:06:26.602632Z     info    starting namespace controller for cluster cluster-1
2020-12-04T20:06:52.103626Z     info    klog    successfully acquired lease istio-system/istio-namespace-controller-election[]
2020-12-04T20:06:52.103895Z     info    starting namespace controller for cluster cluster-0
```

shows we do start ns controller in cluster 0 and 1. but we fail the test at `20:06:50`.

**Comment by howardjohn** [comment]
@stevenctl do you think this is broken or just our timeout is too slow and things are slow since we are deploying 5x istio in parallel?

**Comment by stevenctl** [comment]
It's possible I broke this in #28768 when trying to resolve #28779. Multiple primaries would run the namespace controller for the same remote cluster. I _think_ this only affects the "istiod-less remote" setup which we're running in tests but don't have documented process for. 

Looks like we don't include ConfigMaps in istio-reader role: 
https://github.com/istio/istio/blob/1b0e1cc83754d68ab4fc06f736aec2fe53e42c1f/manifests/charts/base/files/gen-istio-cluster.yaml#L3640

**Comment by howardjohn** [comment]
some more logs: https://storage.googleapis.com/istio-prow/pr-logs/pull/istio_istio/29420/integ-pilot-multicluster-tests_istio/1334979175408209920/build-log.txt

> Looks like we don't include ConfigMaps in istio-reader role, so not sure how this would work at alll:

Very confused why its not deterministic then...

**Comment by howardjohn** [comment]
https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/batch/integ-pilot-multicluster-tests_istio/1335988847711883264

**Comment by stevenctl** [comment]
https://storage.googleapis.com/istio-prow/pr-logs/pull/istio_istio/29352/integ-pilot-multicluster-tests_istio/1335989071104708608/artifacts/pilot-endpointslice-2bb4b62c1b4/_suite_context/istio-state-118212884/cluster-0/istiod-7d7f476b94-d9rgn_discovery.log 

In this instance I'm seeing that istiod in cluster-0 only starts NS controller for cluster-1

**Comment by stevenctl** [comment]
Maybe another cluster is winning leader election but doesn't have permission to do anything. We wouldn't see failures when the primary clusters win the election for their own clusters, or when they win election for remotes, because right now we don't deploy VMs in remotes.

**Comment by stevenctl** [comment]
I think resolving this todo may also fix the issue:

https://github.com/istio/istio/blob/f348ce0f163515d92ceb0b63c7fc9714e66cdd77/pilot/pkg/serviceregistry/kube/controller/multicluster.go#L159

**Comment by stevenctl** [comment]
This happened in ipv6 tests... https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/29619/integ-ipv6-k8s-tests_istio/1338660393626636288

I think this is related to the new VM bootstrap trying to generate files that we push to the VM.

**Comment by stevenctl** [comment]
https://github.com/istio/istio/blob/cf71aad260a6be5194d79a0669b25deb89a850a1/manifests/charts/base/templates/clusterrole.yaml#L138-L148

So when patching for a remote cluster, everything is fine. But if we win the race and try to patch another primary cluster, we will fail. I'm confused how another primary cluster could win the leader-election without permission to mutate the configmap though.

**Comment by stevenctl** [comment]
In #30288 I accidentally made all 3 clusters primary clusters. This issue is very easy to replicate when we have more primary clusters, which is good to know for making sure it's fixed.

**Comment by brian-avery** [comment]
What's the status of this?

**Comment by stevenctl** [comment]
This shouldn't actually affect most users, it's related to external istiod and some things we do in integration tests that wouldn't be done in production. 

The fix would be a bit complicated and isn't a priority for 1.9 as far as I'm concerned.

**Comment by brian-avery** [comment]
Thanks

**Comment by brian-avery** [comment]
@stevenctl revisiting this. We will be carrying this over to 1.11. How is this looking? Thanks.

**Comment by stevenctl** [comment]
this will be fixed with #27420 #33619
